spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -91,112 +108,51 @@ public void setProducerListener(ProducerListener<K, V> producerListener) {
 	}
 
 	@Override
-	public Future<RecordMetadata> send(V data) {
+	public ListenableFuture<SendResult<K, V>> send(V data) {
 		return send(this.defaultTopic, data);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(K key, V data) {
+	public ListenableFuture<SendResult<K, V>> send(K key, V data) {
 		return send(this.defaultTopic, key, data);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(int partition, K key, V data) {
+	public ListenableFuture<SendResult<K, V>> send(int partition, K key, V data) {
 		return send(this.defaultTopic, partition, key, data);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(String topic, V data) {
+	public ListenableFuture<SendResult<K, V>> send(String topic, V data) {
 		ProducerRecord<K, V> producerRecord = new ProducerRecord<>(topic, data);
 		return doSend(producerRecord);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(String topic, K key, V data) {
+	public ListenableFuture<SendResult<K, V>> send(String topic, K key, V data) {
 		ProducerRecord<K, V> producerRecord = new ProducerRecord<>(topic, key, data);
 		return doSend(producerRecord);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(String topic, int partition, V data) {
+	public ListenableFuture<SendResult<K, V>> send(String topic, int partition, V data) {
 		ProducerRecord<K, V> producerRecord = new ProducerRecord<K, V>(topic, partition, null, data);
 		return doSend(producerRecord);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(String topic, int partition, K key, V data) {
+	public ListenableFuture<SendResult<K, V>> send(String topic, int partition, K key, V data) {
 		ProducerRecord<K, V> producerRecord = new ProducerRecord<>(topic, partition, key, data);
 		return doSend(producerRecord);
 	}
 
 	@Override
-	public Future<RecordMetadata> send(Message<?> message) {
+	public ListenableFuture<SendResult<K, V>> send(Message<?> message) {
 		ProducerRecord<K, V> producerRecord = messageToProducerRecord(message);
```

Comment:
Yeah, thank you @sdeleuze !

Will try to study what you, guys, do there some days...